app/controllers/pets_controller.rb

```diff
@@ -45,11 +45,17 @@ def change_application_status
         pet.change_to_pending
       end
       redirect_to "/applications/#{application_id}"
-    else 
+    else
       pet = Pet.find(params[:id])
-      pet.update(approved_applicant: approved_applicant_name)
-      pet.change_to_pending
-      redirect_to "/pets/#{pet.id}"
+      if pet.has_approved_applicant
+        pet.update(approved_applicant: nil)
+        pet.change_to_adoptable
+        redirect_to "/applications/#{application_id}"
+      else
+        pet.update(approved_applicant: approved_applicant_name)
+        pet.change_to_pending
+        redirect_to "/pets/#{pet.id}"
+      end
```

Comment:
Love that we could use this same controller action for unapprove. Maybe at the end, we could look into if this could be refactored at all to make it shorter.